### PR TITLE
ci: add release plugin workflow

### PR DESCRIPTION
**Pull request by CaptainVincent**
As title.

### CODE REVIEW

**Review by hydai — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+            mv ${{ env.output_dir }}/${{ env.output_bin }} ${{ env.output_fmt_bin }}
+            tar -czvf plugin_wasinn_burnrs_${model}.tar.gz ${{ env.output_fmt_bin }}
+          done
+      - name: Upload wasi_nn-burnrs plugin tar.gz package
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          for model in ${{ env.models }}; do
+            mv plugin_wasinn_burnrs_${model}.tar.gz WasmEdge-plugin-wasi_nn-burnrs-${model}-${{ needs.get_version.outputs.version }}-darwin_${{ matrix.arch }}.tar.gz
+            gh release upload ${{ needs.get_version.outputs.version }} WasmEdge-plugin-wasi_nn-burnrs-${model}-${{ needs.get_version.outputs.version }}-darwin_${{ matrix.arch }}.tar.gz --clobber
```
Comment (0yi0) [inline .github/workflows/release.yml:180]: darwin?

**Review by 0yi0 — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (CaptainVincent) [reply .github/workflows/release.yml:180]: Is this an incorrect use of name here? I just followed the [release.yml for WasmEdge](https://github.com/WasmEdge/WasmEdge/blob/1e9bd63e01244023b0acd8d5cc802923f67a0a7d/.github/workflows/release.yml#L193). If there's an error, could you suggest the correct value?

Comment (0yi0) [reply .github/workflows/release.yml:180]: `darwin` are only used for macOS, so I think it should be `asset_tag` here.

Comment (0yi0) [reply .github/workflows/release.yml:180]: Sorry I misread the job name

**Review by 0yi0 — APPROVED** [review]

**Pull request merged**